blazar/plugins/oshosts/host_plugin.py

```diff
@@ -973,6 +973,39 @@ def poll_resource_failures(self):
                 recovered_hosts.extend([host for host in unreservable_hosts
                                         if host['id'] in active_hv_ids])
 
+            aggregates = self.nova.aggregates.list()
+            if aggregates:
+                freepool = next((agg for agg in aggregates if agg.name == "freepool"), None)
```

Comment:
I am concerned this might be slow, and I wonder if there is a better API for it. I see there is `get_details` in the sdk, but not sure what exact versions we are using. https://docs.openstack.org/python-novaclient/latest/reference/api/novaclient.v2.aggregates.html

Reply:
I guess it actually doesn't matter, since we are iterating through the list later in the function.